source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/readonly/ContainerRenderer.java

```diff
@@ -104,6 +108,26 @@ else if ((container = Container.dynamic_cast(baseCardElement)) == null)
             containerView.setBackgroundColor(Color.parseColor(color));
         }
 
+        BackgroundImage backgroundImageProperties = container.GetBackgroundImage();
+        if (backgroundImageProperties != null && !backgroundImageProperties.GetUrl().isEmpty())
```

Comment:
>backgroundImageProperties.GetUrl() [](start = 50, length = 34)

is there any chance the url is null?

Reply:
Just double checked. Since the `url` property is required, having a card with it as `null` will make the `BackgroundImage` object invalid and set the entire object to `null`.

---
In reply to: [256952313](https://github.com/Microsoft/AdaptiveCards/pull/2361#discussion_r256952313) [](ancestors = 256952313)

Reply:
Is it possible to construct a `BackgroundImage` object without an `url` through the object model?

---
In reply to: [257025257](https://github.com/Microsoft/AdaptiveCards/pull/2361#discussion_r257025257) [](ancestors = 257025257,256952313)

Reply:
For the object model, if `url=nullptr`, you get an exception. If `url` is just declared but not instantiated, the empty constructor is called and you get the empty string. 

---
In reply to: [257026918](https://github.com/Microsoft/AdaptiveCards/pull/2361#discussion_r257026918) [](ancestors = 257026918,257025257,256952313)